_field-types/supported-field-types/star-tree.md

```diff
@@ -0,0 +1,177 @@
+
+```json
+PUT logs
+{
+  "settings": {
+    "index.number_of_shards": 1,
+    "index.number_of_replicas": 0,
+    "index.composite_index": true
+  },
+  "mappings": {
+    "composite": {
+      "startree1": {
+        "type": "star_tree",
+        "config": {
+          "max_leaf_docs": 10000,
+          "skip_star_node_creation_for_dimensions": [
+            "port"
+          ],
+          "ordered_dimensions": [
+            {
+              "name": "status"
+            },
+            {
+              "name": "port"
+            }
+          ],
+          "metrics": [
+            {
+              "name": "request_size",
+              "stats": [
+                "sum",
+                "value_count",
+                "min",
+                "max"
+              ]
+            },
+            {
+              "name": "latency",
+              "stats": [
+                "sum",
+                "value_count",
+                "min",
+                "max"
+              ]
+            }
+          ]
+        }
+      }
+    },
+    "properties": {
+      "status": {
+        "type": "integer"
+      },
+      "port": {
+        "type": "integer"
+      },
+      "request_size": {
+        "type": "integer"
+      },
+      "latency": {
+        "type": "scaled_float",
+        "scaling_factor": 10
+      }
+    }
+  }
+}
+```
+In the above example, for `startree1` , we will create an associated Star Tree index. Currently only `one` Star Tree index can be created per index. Support for multiple Star Trees is a work in progress and will be released in the future. <br/>
+
+## Star tree mapping parameters
+Specify Star Tree configuration under `config` section. All parameters are final and cannot be modified without reindexing documents.
+
+### Ordered dimensions
+The `ordered_dimensions` are fields based on which the metrics will be aggregated in a Star Tree index. Star Tree index will be picked for querying only if all the fields in the query are part of the `ordered_dimensions`.
+- The order of dimensions matter and you can define the dimensions ordered from the highest cardinality to the lowest cardinality for efficient storage and query pruning. 
+- Avoid high cardinality fields as dimensions as it'll affect storage space, indexing throughput and query performance adversely.
+- Currently, supported fields for `ordered_dimensions` are of [numeric field types](https://opensearch.org/docs/latest/field-types/supported-field-types/numeric/) with the exception of `unsigned_long`(see [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/15231)). 
+  - Support for other field_types such as `keyword`, `ip` is currently work in progress and will be released in future versions (see [GitHub issue](https://github.com/opensearch-project/OpenSearch/issues/16232))
+- Minimum of `2` and maximum of `10` dimensions are supported per Star Tree index.
```

Comment:
Doesn't it support just one dimension?

Reply:
Right now we support minimum of 2 dimensions. 

I too thought if really 2 dimensions are needed especially with date field. Maybe we can remove the restriction in next possible release.